plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -90,22 +87,48 @@ export function WriteItForMeButtonContainer() {
 
 	useEffect( () => {
 		const title = titleEl.current;
-		const titleKeyupHandler = ( e: KeyboardEvent ) =>
-			setProductTitle( ( e.target as HTMLInputElement ).value || '' );
-
+		const titleKeyupHandler = ( e: KeyboardEvent ) => {
+			setTitleIsGenerated( false );
+			setProductTitle(
+				( e.target as HTMLInputElement ).value.trim() || ''
+			);
+		};
 		title?.addEventListener( 'keyup', titleKeyupHandler );
 
+		// This observer is used to detect when the title is generated.
+		const titleObserver = new MutationObserver( ( mutations ) => {
```

Comment:
A bit of logic to detect if the title was generated. Probably the best way to do so short of something more elaborate such as slot-fills to manage collective state.

Reply:
I would like to see us try to avoid making things more complicated, what about a couple of other options?

- Add in the "title generation" prompt a requirement that the output has at-least 20 characters to meet the requirement for the description. One pro here is that any title generated will have enough length to immediately generate a description. Another pro is that we set an opinionated feature that has proven to be beneficial to stores (longer, descriptive titles). A downside might be that titles become pretty elaborate with languages like Chinese/Japanese but it didn't seem egregious to me when testing.
- Get rid of/lower the minimum character requirement. The pro here is that it makes it easier to generate a description. The con is that the output is going to be of lower quality given that there is so little information to pass into the prompt. 🤷 

My preference is that we should be okay with an opinionated approach if the data backs it up.

Reply:
Yeah, I would agree that this is overly complex. It does feel that we've haven't arrived at an ideal in how to implement this, but starting with a simpler approach such as your PR sounds good to me 👍

Reply:
Removed in 5feffabd85bc1adcecde6eed411c05c1033ec397